boa/src/builtins/string/mod.rs

```diff
@@ -1165,6 +1166,81 @@ impl String {
         }
     }
 
+    /// String.prototype.split()
+    ///
+    /// The `split()` method divides a String into an ordered list of substrings, puts these substrings into an array, and returns the array.
+    ///
+    /// The division is done by searching for a pattern; where the pattern is provided as the first parameter in the method's call.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.split
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split
+    pub(crate) fn split(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let string = this.to_string(context)?;
+
+        let separator = if args.is_empty() {
+            None
+        } else {
+            let arg0 = args
+                .get(0)
+                .expect("Failed to get argument for String method");
```

Comment:
I think this this would be more idiomatic using `args.get(0).and_then(|arg0| { ... });`. It's true that the `Err` variant can make it a bit more complex, but maybe returning an `Option<Result>` and then calling `transpose()` on it could help.

Reply:
Good idea, I'll remove the outer if-else with this.